project/driver_controls/src/driver_state.c

```diff
@@ -0,0 +1,50 @@
+#include <stdio.h>
+
+#include "driver_state.h"
+#include "log.h"
+#define MAX_FSMS 10
+
+typedef bool (*TransitionCheck)(Event *e);
+
+static FSM *s_driver_fsms[MAX_FSMS];
+static bool s_fsm_approval[MAX_FSMS];
+
+static uint8_t s_active_fsms = 0;
+
+static bool prv_get_permit(Event *e) {
+  bool transitioned;
+
+  for (uint8_t i = 0; i < s_active_fsms; i++) {
+    s_driver_fsms[i]->current_state->output(s_driver_fsms[i], e, NULL);
```

Comment:
Do not poke into FSM internals. I would suggest setting the context to storage for the function pointer  instead. The FSM's output function could change the function pointer. The difference between what you had before and what I'm suggesting is that you don't need to store the FSMs or poke into their internals. Instead, since you passed in the FSM's context, you can iterate over your storage array of function pointers.